### PR TITLE
Expand teraslice cli functionality

### PR DESCRIPTION
**Pull request by jeffmontagna**
These changes expands the functionality of the cli tool to interface with teraslice clusters without the needing the job files
- Sub-commands are split out into individual files
- Config file has been added and is stored in `~/.teraslice/earl.yaml
- Clusters can be identified with an alias, which is saved in the config file 
- `cjob` commands right now only work with ids specified on the command line or running on the cluster.
- Plan on renaming command from tjm to earl.

### CODE REVIEW

**Comment by kstaken** [comment]
package.json has problems with comma placement. I manually fixed that to try to test then yarn says the following and gives you a bunch of versions to choose from.

```
Couldn't find any versions for "prompts" that matches "^1.1.11"
? Please choose a version of "prompts" from this list: 1.1.1
```

File: packages/teraslice-cli/package.json [hunk 1]
```diff
@@ -1,47 +1,56 @@
...
+    "teraslice-client-js": "^0.3.0",
+    "yargs": "^12.0.2",
+    "teraslice-client-js": "^0.1.2",
+    "node-yaml": "^3.1.1",
+    "easy-table": "^1.1.1",
+    "mkdirp": "^0.5.1",
+    "process": "^0.11.10",
+    "syncprompt": "^2.0.0",
+    "terminal-kit": "^1.26.0",
+    "tty-table": "^2.6.8",
```
Comment (ciorg) [inline packages/teraslice-cli/package.json]: extra comma

**Comment by kstaken** [comment]
Lots of issues with missing params.

```
teraslice-cli ex list -c
/Users/kstaken/projects/opensource/teraslice/packages/teraslice-cli/node_modules/yargs/yargs.js:1144
      else throw err
           ^

TypeError: inUrl.indexOf is not a function
    at _urlCheck (/Users/kstaken/projects/opensource/teraslice/packages/teraslice-cli/cmds/lib/config.js:149:19)
    at getClusterHost (/Users/kstaken/projects/opensource/teraslice/packages/teraslice-cli/cmds/lib/config.js:90:26)
    at Object.returnConfigData (/Users/kstaken/projects/opensource/teraslice/packages/teraslice-cli/cmds/lib/config.js:58:81)
    at Object.exports.handler (/Users/kstaken/projects/opensource/teraslice/packages/teraslice-cli/cmds/ex/list.js:23:34)
    at Object.runCommand (/Users/kstaken/projects/opensource/teraslice/packages/teraslice-cli/node_modules/yargs/lib/command.js:238:44)
    at Object.parseArgs [as _parseArgs] (/Users/kstaken/projects/opensource/teraslice/packages/teraslice-cli/node_modules/yargs/yargs.js:1059:30)
    at Object.runCommand (/Users/kstaken/projects/opensource/teraslice/packages/teraslice-cli/node_modules/yargs/lib/command.js:198:45)
    at Object.parseArgs [as _parseArgs] (/Users/kstaken/projects/opensource/teraslice/packages/teraslice-cli/node_modules/yargs/yargs.js:1059:30)
    at Object.get [as argv] (/Users/kstaken/projects/opensource/teraslice/packages/teraslice-cli/node_modules/yargs/yargs.js:1000:21)
    at Object.<anonymous> (/Users/kstaken/projects/opensource/teraslice/packages/teraslice-cli/index.js:5:18)
```

```
earl cluster alias
/Users/kstaken/projects/opensource/teraslice/packages/teraslice-cli/node_modules/yargs/yargs.js:1144
      else throw err
           ^

TypeError: Cannot read property 'indexOf' of undefined
    at valid (/Users/kstaken/projects/opensource/teraslice/packages/teraslice-cli/cmds/lib/shorthand.js:8:17)
    at Object.parse (/Users/kstaken/projects/opensource/teraslice/packages/teraslice-cli/cmds/lib/shorthand.js:40:13)
    at Object.returnConfigData (/Users/kstaken/projects/opensource/teraslice/packages/teraslice-cli/cmds/lib/config.js:43:41)
    at Object.exports.handler (/Users/kstaken/projects/opensource/teraslice/packages/teraslice-cli/cmds/cluster/alias.js:36:40)
    at Object.runCommand (/Users/kstaken/projects/opensource/teraslice/packages/teraslice-cli/node_modules/yargs/lib/command.js:238:44)
    at Object.parseArgs [as _parseArgs] (/Users/kstaken/projects/opensource/teraslice/packages/teraslice-cli/node_modules/yargs/yargs.js:1059:30)
    at Object.runCommand (/Users/kstaken/projects/opensource/teraslice/packages/teraslice-cli/node_modules/yargs/lib/command.js:198:45)
    at Object.parseArgs [as _parseArgs] (/Users/kstaken/projects/opensource/teraslice/packages/teraslice-cli/node_modules/yargs/yargs.js:1059:30)
    at Object.get [as argv] (/Users/kstaken/projects/opensource/teraslice/packages/teraslice-cli/node_modules/yargs/yargs.js:1000:21)
    at Object.<anonymous> (/Users/kstaken/projects/opensource/teraslice/packages/teraslice-cli/index.js:5:18)
```

**Comment by kstaken** [comment]
I setup an alias with the port then tried to use it.

```TypeError [ERR_INVALID_URL]: Invalid URL: http://localhost:5678:5678```

**Comment by kstaken** [comment]
```
earl jobs run
Cannot read property 'id' of undefined
```

```
earl jobs run data_generator_kafka.json
/Users/kstaken/projects/opensource/teraslice/node_modules/yargs/yargs.js:1144
      else throw err
           ^

TypeError: Cannot read property 'cluster_manager_type' of undefined
    at Object.returnConfigData (/Users/kstaken/projects/opensource/teraslice/packages/teraslice-cli/cmds/lib/config.js:60:40)
    at Object.exports.handler (/Users/kstaken/projects/opensource/teraslice/packages/teraslice-cli/cmds/jobs/run.js:17:34)
    at Object.runCommand (/Users/kstaken/projects/opensource/teraslice/node_modules/yargs/lib/command.js:238:44)
    at Object.parseArgs [as _parseArgs] (/Users/kstaken/projects/opensource/teraslice/node_modules/yargs/yargs.js:1059:30)
    at Object.runCommand (/Users/kstaken/projects/opensource/teraslice/node_modules/yargs/lib/command.js:198:45)
    at Object.parseArgs [as _parseArgs] (/Users/kstaken/projects/opensource/teraslice/node_modules/yargs/yargs.js:1059:30)
    at Object.get [as argv] (/Users/kstaken/projects/opensource/teraslice/node_modules/yargs/yargs.js:1000:21)
    at Object.<anonymous> (/Users/kstaken/projects/opensource/teraslice/packages/teraslice-cli/index.js:5:18)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
```

```
earl job stop data_generator_kafka.json
/Users/kstaken/projects/opensource/teraslice/node_modules/yargs/yargs.js:1144
      else throw err
           ^

TypeError: terasliceClient.jobs.wrap(...).spec is not a function
    at Object.alreadyRegisteredCheck (/Users/kstaken/projects/opensource/teraslice/packages/teraslice-cli/cmds/cmd_functions/functions.js:23:70)
    at Object.exports.handler (/Users/kstaken/projects/opensource/teraslice/packages/teraslice-cli/cmds/job/stop.js:18:25)
    at Object.runCommand (/Users/kstaken/projects/opensource/teraslice/node_modules/yargs/lib/command.js:238:44)
    at Object.parseArgs [as _parseArgs] (/Users/kstaken/projects/opensource/teraslice/node_modules/yargs/yargs.js:1059:30)
    at Object.runCommand (/Users/kstaken/projects/opensource/teraslice/node_modules/yargs/lib/command.js:198:45)
    at Object.parseArgs [as _parseArgs] (/Users/kstaken/projects/opensource/teraslice/node_modules/yargs/yargs.js:1059:30)
    at Object.get [as argv] (/Users/kstaken/projects/opensource/teraslice/node_modules/yargs/yargs.js:1000:21)
    at Object.<anonymous> (/Users/kstaken/projects/opensource/teraslice/packages/teraslice-cli/index.js:5:18)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Function.Module.runMain (module.js:693:10)
    at startup (bootstrap_node.js:191:16)
    at bootstrap_node.js:612:3
```

**Comment by kstaken** [comment]
Let's remove `tjm` from the command names in package.json so that we can have the old tool installed in parallel.

**Comment by kstaken** [comment]
Also let's go ahead and remove all the legacy commands out of this and just focus on making this work correctly with the new commands for all use cases.

**Comment by kstaken** [comment]
`alias` should be `aliases` since everything else is plural now.

**Pull request merged**